### PR TITLE
Allow using browserlistsrc for targets

### LINKED ISSUE #1141: Allow browserslistsrc file to configure babel targets
A while back I brought this up: https://github.com/neutrinojs/neutrino/issues/229

It is desirable to have a browserslist file so various tools can pick use it (autoprefixer, stylelint, eslint-plugin-compat, and of course, @babel/preset-env).

`@babel/preset-env` supports this out of the box: https://babeljs.io/docs/en/babel-preset-env#browserslist-integration, however, since Neutrino passes targets directly, I'm assuming it will take those values and not read them from the file.

I'm not sure if this is a situation where we can use `neutrinorc` as the single source of truth, as `.browserslistrc` files, I believe are just plain text (no js).

### PR DESCRIPTION
**Pull request by timkelty**
Resolves #1141

Allows users to pass `false` to `options.targets` or `options.targets.browsers` to have babel-env use a `.browserlistsrc` file.

### CODE REVIEW

File: packages/web/index.js [hunk 1]
```diff
@@ -90,7 +94,7 @@ module.exports = (neutrino, opts = {}) => {
       'last 2 iOS versions'
     ];
   }
-
+
```
Comment (eliperelman) [inline packages/web/index.js]: Can you revert the change to this line?

File: packages/web/index.js [hunk 2]
```diff
@@ -80,7 +80,11 @@ module.exports = (neutrino, opts = {}) => {
     };
   }
 
-  if (!options.targets.node && !options.targets.browsers) {
+  if (options.targets === false) {
+    options.targets = {};
+  } else if (options.targets.browsers === false) {
```
[the remark this answers is not in the document]
Comment (timkelty) [reply packages/web/index.js]: Yeah - I did it this way because it seemed like `options.targets` had other props besides browsers that maybe you'd want to set, but still use the browerslistrc. Not totally sure of that's valid though, will have to check when I'm back on a machine: https://babeljs.io/docs/en/babel-preset-env#options

File: packages/web/index.js [hunk 3]
```diff
@@ -80,7 +80,11 @@ module.exports = (neutrino, opts = {}) => {
     };
   }
 
-  if (!options.targets.node && !options.targets.browsers) {
+  if (options.targets === false) {
```
[the remark this answers is not in the document]
Comment (timkelty) [reply packages/web/index.js]: Will do

File: packages/web/index.js (hunk 1 above)
Comment (timkelty) [reply packages/web/index.js]: Whups

Comment (timkelty) [reply packages/web/index.js]: Whups

**Comment by timkelty** [comment]
@edmorley I think you're right – we only need to support `options.targets` as `false`, not `options.targets.browsers`. Updated PR.

**Comment by timkelty** [comment]
Actually, if we set the defaults up-top like this, we don't even need to account for `false`, and have users just pass `{}` if they want to overrides defaults. Thoughts?

**Comment by edmorley** [comment]
>  Seem simpler. Thoughts?

The preset default options are merged with the ones the user passes using deepmerge, so I think the new implementation will prevent people from overriding. Some tests would confirm and likely be useful edither way :-)

**Comment by timkelty** [comment]
> The preset default options are merged with the ones the user passes using deepmerge, so I think the new implementation will prevent people from overriding

@edmorley right you were :) Reverted that and added some tests.
